### PR TITLE
[WIP] feat: support management of azure feeds

### PR DESCRIPTION
**Pull request by C123R**
This is just a draft version of new feature of supporting azure feeds(#280)

I will fill required information here once I am ready with it.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,7 +12,9 @@ require (
 	github.com/microsoft/azure-devops-go-api/azuredevops v0.0.0-20200327121006-543de4815ec2
 	github.com/stretchr/testify v1.3.0
 	golang.org/x/crypto v0.0.0-20200427165652-729f1e841bcc
+	golang.org/x/lint v0.0.0-20200302205851-738671d3881b // indirect
```
Comment (nmiodice) [inline go.mod:15]: I know this is a WIP, but please run `go mod vendor` prior to submitting your final PR to ensure that all changes to these files are represented in the `vendor` directory.

**Comment by C123R** [comment]
Looking for the suggestion, @nmiodice maybe you could help me out.

I am a bit confused with the lifecycle of Azure Artifacts feed.. as currently, the [deletion](https://docs.microsoft.com/en-us/rest/api/azure/devops/artifacts/feed%20%20management/delete%20feed?view=azure-devops-rest-6.0) moves the feed into the recycle bin for 30 days with the option of restoring.
I guess [permanent deletion](https://docs.microsoft.com/en-us/rest/api/azure/devops/artifacts/feed%20recycle%20bin/permanent%20delete%20feed?view=azure-devops-rest-6.0) is also possible but still same feed name will be available to reuse only after 15 mins.

This leads to the confusion, how should we deal with terraform update operation where recreation is required:

For example:
```
resource "azuredevops_feed" "myfeed" {
  name                          = "myfeed"
  project_name                  = azuredevops_project.project.project_name <-----
  description                   = "Sample terraform feed"
  is_badge_enabled              = true
  hide_deleted_package_versions = true
}
```

Changing project_name should force recreation as feeds cant be moved across projects.

ERROR: `Error creating feed: A feed with name 'myfeed' exists in feed recycle bin. The name will be available soon after it is permanently deleted from the feed recycle bin.`

**Comment by nmiodice** [comment]
Hmm, I was not aware of this behavior. The lease on the feed name is, from an automation perspective, (Terraform, CLI, etc...) a pretty bad experience but not one that we can work around because it is a service side behavior.

@tedchamb do you know a good place to request information from the ADO team? Is there a developer community against which we can create a bug and/or ask a question about this behavior?

It seems like the options we have for the implementation are the following:

1. A `destroy` will move the feed into the Recycling Bin
2. A `destroy` will move the feed into the Recycling Bin and then permanently delete it from the Recycling Bin
3. Do not implement a `destroy`

I don't feel that I have enough experience using feeds in practice to have an informed opinion on this, except that I'd prefer to avoid going with option `3`.

@xuzhang3 @mozehgir @EliiseS @rguthriemsft @tmeckel - my trusted DevOps experts - do you have any perspective on this?

**Comment by EliiseS** [comment]
Throwing caution to the wind, but my best out of box thinking for this with permanent deletion:
1) Wait until it becomes available after permanent deletion (15 minute wait time)
2) Have a user assign a second "failover" name that'll be used for the new feed in the event of deletion 
3) Give a user a choice between the two above
4) In the event of deletion, throw an error that says a feed can't be created until x minutes or a new name must be provided

These all feel a like a version of bad, but unless there's additional information, I think waiting would be the most viable?

**Comment by C123R** [comment]
> D) In the event of deletion, throw an error that says a feed can't be created until x minutes or a new name must be provided

I like this option for now, as currently, we need recreation or force replacement only in case when project reference is getting changed, throwing an error suggested by you totally makes sense for me.

Another thing which I just noticed: Currently [azdo go sdk](https://github.com/microsoft/azure-devops-go-api) doesn't support [permanent deletion](https://docs.microsoft.com/en-us/rest/api/azure/devops/artifacts/feed%20recycle%20bin/permanent%20delete%20feed?view=azure-devops-rest-6.0). But for sure I could create a pull request for that.

Btw thanks for your suggestions @nmiodice @EliiseS

**Comment by tedchamb** [comment]
@nmiodice , the feedback section at the bottom of the REST doc pages is one place you can submit issues and/or questions regarding the REST APIS.  For `Recycle Bin` the page is here:

https://docs.microsoft.com/en-us/rest/api/azure/devops/artifacts/recycle%20%20bin?view=azure-devops-rest-6.0

I also sent you a DM with an internal link, that has contact information for the various areas.  I am not sure how up to date it is.

**Comment by nmiodice** [comment]
I like option `D` as well

**Comment by EliiseS** [comment]
@C123R, we have migrate this repository to terraform-provider here: https://github.com/terraform-providers/terraform-provider-azuredevops. Could you resubmit your PR to the there?

I'll be closing this PR here.

**Pull request closed without merging**